vendor: bump Pebble to 8c3bd06debb5

jbowens opened the pull request:
```
8c3bd06d vfs: export ErrUnsupported
b8b89851 tool: add manifest summarize command
8ac9797e internal/manifest: Move Atomic field in the FileMetaData struct to the top.
b8f537d8 compaction: remove nonzero-seqnum splitting logic
5c39ff26 benchmarks: infrastructure to set up file system benchmarks
```

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/67747)
<!-- Reviewable:end -->

bananabrick reviewed (approved):
> Reviewed 4 of 4 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/67747)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @itsbilal and @sumeerbhola)



<!-- Sent from Reviewable.io -->

jbowens committed:
vendor: bump Pebble to 8c3bd06debb5

8c3bd06d vfs: export ErrUnsupported
b8b89851 tool: add manifest summarize command
8ac9797e internal/manifest: Move Atomic field in the FileMetaData struct to the top.
b8f537d8 compaction: remove nonzero-seqnum splitting logic
5c39ff26 benchmarks: infrastructure to set up file system benchmarks

Release note: None

jbowens reviewed (commented):
TFTR!

bors r=bananabrick

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/67747#-:-Mf4pqG93NQXlFUVT_31:b-q5qotx)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @bananabrick, @itsbilal, and @sumeerbhola)



<!-- Sent from Reviewable.io -->
